### PR TITLE
Filter NRES and FLOYDS data

### PR DESCRIPTION
**Pull request by cmccully**
Added the ability to filter out non-imaging data from banzai pipeline. This allows us to run a banzai-nres pipeline in parallel.

### CODE REVIEW

File: banzai/preview.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+from banzai import dbs
+from banzai.utils import file_utils
+from banzai import logs
+from banzai.utils.image_utils import image_passes_criteria
+
+logger = logs.get_logger(__name__)
+
+
```
Comment (turnerm) [inline banzai/preview.py]: Should `PreviewModeListener` be moved here?

File: banzai/main.py [hunk 2]
```diff
@@ -45,6 +49,10 @@
                   astrometry.WCSSolver,
```
Comment (turnerm) [inline banzai/main.py]: `ordered_stages` should probably be all caps too. Your favourite!

Comment (cmccully) [reply banzai/main.py]: You are of course correct.

File: banzai/preview.py (hunk 1 above)
Comment (cmccully) [reply banzai/preview.py]: Indeed.

Comment (cmccully) [reply banzai/preview.py]: Oh. Right. Circular imports make this really complicated. This will be refactored once we merge end of night and preview modes.

**Pull request merged**